packages/module-utils/src/generate-dts-sourcemaps.ts

```diff
@@ -0,0 +1,256 @@
+    return [createLinePartMapping(dtsOffset, srcLine, srcCol)];
+}
+
+type Position = {
+    line: number;
+    column: number;
+};
+
+function createStateLineMapping(
+    stateTokens: ClassStateToken[],
+    srcClassName: string,
+    lastSrcLocation: Position,
+    meta: StylableMeta
+) {
+    const res: { mapping: LinePartMapping[]; lastSrcPosition: Position } = {
+        mapping: [],
+        lastSrcPosition: lastSrcLocation,
+    };
+    let prevDtsColumn = 0;
+    for (const stateToken of stateTokens) {
+        let stateSourcePosition: Position | undefined;
+
+        meta.rawAst.walkRules(`.${srcClassName}`, (rule) => {
+            return rule.walkDecls(valueMapping.states, (decl) => {
+                if (decl.source && decl.source.start)
+                    stateSourcePosition = {
+                        line: decl.source.start.line - 1,
+                        column: decl.source.start.column - 1,
+                    };
+                return false;
+            });
+        });
+
+        if (stateSourcePosition) {
+            res.mapping.push(
+                createLinePartMapping(
+                    stateToken.stateName.column - prevDtsColumn,
+                    stateSourcePosition.line - res.lastSrcPosition.line,
+                    stateSourcePosition.column - res.lastSrcPosition.column
+                )
+            );
+
+            prevDtsColumn = stateToken.stateName.column;
+            res.lastSrcPosition = {
+                line: stateSourcePosition.line,
+                column: stateSourcePosition.column,
+            };
+        }
+    }
+
+    return res;
+}
+
+function findTokenForLine(line: number, dstTokens: TokenizedDtsEntry[]) {
+    for (const typedTokens of dstTokens) {
+        for (const token of typedTokens.tokens) {
+            if ('line' in token) {
+                if (token.line === line) {
+                    return { type: typedTokens.type, token: token };
+                }
+            } else if (
+                token.classStates &&
+                token.classStates.length &&
+                token.classStates[0].stateName.line === line
+            ) {
+                return { type: typedTokens.type, token: token };
+            }
+        }
+    }
+
+    return;
+}
+
+function getClassSourceName(targetName: string, classTokens: ClassesToken): string | undefined {
+    for (const classToken of classTokens.tokens) {
+        if (classToken.outputValue?.value === targetName) {
+            return classToken.value;
+        }
+    }
+
+    return;
+}
+
+export function generateDTSSourceMap(dtsContent: string, meta: StylableMeta) {
+    const tokens = tokenizeDTS(dtsContent);
+    const mapping: Record<number, LineMapping> = {};
+    const lines = dtsContent.split('\n');
+    let lastSrcLocation: Position = { line: 0, column: 0 };
+
+    for (const dtsLine of lines.keys()) {
+        const resToken = findTokenForLine(dtsLine, tokens);
```

Comment:
this makes an assumption of the way the `d.ts` is structured (1 symbol at a line) - maybe add a comment explaining how this function/util works